src/app/features/courses/courses.component.html

```diff
@@ -1 +1,33 @@
-<p>courses works!</p>
+<div *ngIf="coursesList$ | async as coursesList; else noCourses">
+  <div *ngIf="coursesList.length > 0; else noCourses">
+    <div class="courses__header">
+      <app-search (search)="onSearch($event)"></app-search>
+      <app-button
+        (click)="redirectToAddCourse()"
```

Comment:
Where give a have related to the event gives a flexibility when action will be changed.